scalalib/overrides-2.12/scala/reflect/Manifest.scala

```diff
@@ -87,29 +86,14 @@ object ManifestFactory {
   def valueManifests: List[AnyValManifest[_]] =
     List(Byte, Short, Char, Int, Long, Float, Double, Boolean, Unit)
 
-  def Byte: AnyValManifest[Byte] = ByteManifest
-  def Short: AnyValManifest[Short] = ShortManifest
-  def Char: AnyValManifest[Char] = CharManifest
-  def Int: AnyValManifest[Int] = IntManifest
-  def Long: AnyValManifest[Long] = LongManifest
-  def Float: AnyValManifest[Float] = FloatManifest
-  def Double: AnyValManifest[Double] = DoubleManifest
-  def Boolean: AnyValManifest[Boolean] = BooleanManifest
-  def Unit: AnyValManifest[Unit] = UnitManifest
-  def Any: Manifest[scala.Any] = AnyManifest
-  def Object: Manifest[java.lang.Object] = ObjectManifest
-  def AnyRef: Manifest[scala.AnyRef] = Object.asInstanceOf[Manifest[scala.AnyRef]]
-  def AnyVal: Manifest[scala.AnyVal] = AnyValManifest
-  def Null: Manifest[scala.Null] = NullManifest
-  def Nothing: Manifest[scala.Nothing] = NothingManifest
-
   private object ByteManifest extends AnyValManifest[scala.Byte]("Byte") {
     def runtimeClass = java.lang.Byte.TYPE
     override def newArray(len: Int): Array[Byte] = new Array[Byte](len)
     override def newWrappedArray(len: Int): WrappedArray[Byte] = new WrappedArray.ofByte(new Array[Byte](len))
     override def newArrayBuilder(): ArrayBuilder[Byte] = new ArrayBuilder.ofByte()
     private def readResolve(): Any = Manifest.Byte
   }
+  def Byte: AnyValManifest[Byte] = ByteManifest
```

Comment:
That's because as a `def`, the optimizer can inline it, see the precise type `ByteManifest$`, and from there inline the methods such as `runtimeClass`. This is part of the Scala.js-specific optimizations we applied on `ClassTag` in https://github.com/scala-js/scala-js/pull/1669